The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/5009370856/jobs/8978235080) [(plain)](https://github.com/rust-lang/rust/commit/5f53341f924a101f0c4c3e50ef99767114f15fa3/checks/13568310462/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
...................

failures:

---- src/option.rs - option::Option<T>::map_or_else (line 1162) stdout ----
error[E0277]: the `?` operator can only be used in a function that returns `Result` or `Option` (or another type that implements `FromResidual`)
  |
  |
3 | fn main() { #[allow(non_snake_case)] fn _doctest_main_src_option_rs_1162_0() {
  |                                      --------------------------------------- this function should return `Result` or `Option` to accept `?`
...
6 |    .map_or_else(|| std::fs::read_to_string("/etc/someconfig.conf"), Ok)?
  |                                                                        ^ cannot use the `?` operator in a function that returns `()`
  |
  = help: the trait `FromResidual<Result<Infallible, std::io::Error>>` is not implemented for `()`

error[E0277]: the `?` operator can only be used in a function that returns `Result` or `Option` (or another type that implements `FromResidual`)
  |
  |
3 | fn main() { #[allow(non_snake_case)] fn _doctest_main_src_option_rs_1162_0() {
  |                                      --------------------------------------- this function should return `Result` or `Option` to accept `?`
7 |    .parse()?;
  |            ^ cannot use the `?` operator in a function that returns `()`
  |
  |
  = help: the trait `FromResidual<Result<Infallible, _>>` is not implemented for `()`
error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0277`.
Couldn't compile the test.
Couldn't compile the test.

failures:
    src/option.rs - option::Option<T>::map_or_else (line 1162)

test result: FAILED. 4113 passed; 1 failed; 41 ignored; 0 measured; 0 filtered out; finished in 48.05s

error: doctest failed, to rerun pass `-p core --doc`
```

</details>

--- a/library/core/src/option.rs
+++ b/library/core/src/option.rs
@@ -1138,7 +1138,7 @@ impl<T> Option<T> {
     /// Computes a default function result (if none), or
     /// applies a different function to the contained value (if any).
     ///
-    /// # Examples
+    /// # Basic examples
     ///
     /// ```
     /// let k = 21;
@@ -1148,6 +1148,25 @@ impl<T> Option<T> {
     ///
     /// let x: Option<&str> = None;
     /// assert_eq!(x.map_or_else(|| 2 * k, |v| v.len()), 42);
+    /// ```
+    ///
+    /// # Handling a Result-based fallback
+    ///
+    /// A somewhat common occurrence when dealing with optional values
+    /// in combination with [`Result<T, E>`] is the case where one wants to invoke
+    /// a fallible fallback if the option is not present.  This example
+    /// parses a command line argument (if present), or the contents of a file to
+    /// an integer.  However, unlike accessing the command line argument, reading
+    /// the file is fallible, so it must be wrapped with `Ok`.
+    ///
+    /// ```no_run
+    /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
+    /// let v: u64 = std::env::args()
+    ///    .nth(1)
+    ///    .map_or_else(|| std::fs::read_to_string("/etc/someconfig.conf"), Ok)?
+    ///    .parse()?;
+    /// #   Ok(())
+    /// # }
     /// ```
     #[inline]
     #[stable(feature = "rust1", since = "1.0.0")]

